### PR TITLE
S19 t1/cancelling meetings frontend

### PR DESCRIPTION
**Pull request by catai9**
**Scrum Board Ticket:** S19 T1: Create the front end UI for cancelling any meetings

**Story:** Story 19 Meeting Cancellation

**Description:** Added a button to allow any user to cancel any of their upcoming meetings. Upon clicking of the button, a confirmation message will also appear to ensure that the user does indeed want to cancel their meetings. Please note that the term **appointment** is used in the user interface and is synonymous with the term **meeting**. Integrating this functionality with the backend is NOT in scope for this ticket and will be implemented in S19 T3. 

**Testing Criteria (how I tested the code):** 
1. Ran the code locally and verified that the Cancel button was available next to each upcoming appointment for both student and worker users (see Figure 1). I also clicked the button and ensured that a cancellation confirmation popup rendered as expected (see Figure 2). When the _No_ button is clicked, the appointment cancellation confirmation will disappear. However, if the _Yes_ button is clicked, an alert box will appear to let the user know that the appointment has not been cancelled (as it is not yet integrated the feature with the backend). When the user clicks the _Ok_ button, all of the popups will disappear and they will be returned to the home page. 
- STUDENT email: johndoe@gmail.com, password: a1234
- WORKER email: joshuabrooks@gmail.com, password: j1234

image
**Figure 1.** Screenshot for the updated home page for a student user.

image
**Figure 2.** Screenshot of the Appointment Cancellation Confirmation popup.

image
**Figure 3.** Screenshot of the temporary alert message shown to the user when they confirm the appointment cancellation (will be adjusted once the feature is properly integrated with the back end). 

2. Ran the unit tests in the front end and ensured 100% file coverage (and passing tests) for the `AppointmentList.js` component. Also checked to make sure that the `EmergencyDayCancellation.js`, `LogIn.js` and `Title.js` components still have 100% code coverage and the `CheckboxApplication.js` and `LogInForm.js` have above 95% coverage similar to the master branch. I also checked to make sure all tests are passing. The other tests are not set up yet (in the scope of another ticket) and thus the lower file coverage for the other front end files is expected.
3. Ran the automated server test codes and ensured that there was 100% file coverage (and passing tests) for all the unit tests.

**How Reviewers can Test Code:**
1. Run the code locally and ensure no complications when starting up (run the command `npm run-script start:dev` in the root folder OR run the command `npm start` if your node and npm versions are above 12.10.0 for node and 6.10.3 for npm).
a. Use student credentials such as `username: johndoe@gmail.com and password: a1234` and check to make sure that there is a Cancel button next to each upcoming appointment for the student. 
b. Use worker credentials such as `username: joshuabrooks@gmail.com and password: j1234` and check to make sure that there is a Cancel button next to each upcoming appointment for the student. 
c. For either or both users, click the button and ensure that the response is as expected (check above screenshots for the expected results).
2. Run the automated tests locally and ensure that all tests pass and have 100% file coverage. 
a. Navigate inside the `server` folder and run `npm test`. 
b. Navigate inside the `react-ui` folder and run `npm test`. All tests should pass and the `EmergencyDayCancellation.js`, `AppointmentList.js`, `LogIn.js` and `Title.js` components should have 100% file coverage whilst the `CheckboxApplication.js` and `LogInForm.js` have above 95% coverage (the rest are not set up yet and are in scope for another ticket).

### CODE REVIEW

File: react-ui/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "node-sass": "^4.14.1",
     "query-string": "^6.13.6",
     "react": "^16.13.1",
+    "react-confirm-alert": "^2.6.2",
```
Comment (catai9) [inline react-ui/package.json:15]: Please note that a new package was installed so make sure to run `npm install` inside the `react-ui` folder to properly run and test the code locally.

Comment (sehersarin) [reply react-ui/package.json:15]: Thank you for the heads up - will do that!

**Comment by sehersarin** [comment]
This is a minor bug that I found (not super important to fix now but can be something the team looks at in the future.) When I view the UI using Safari the cancel button text is a little off and can't be seen correctly. (Please view screenshot below.) 

Screen Shot 2020-11-26 at 9 58 19 PM

However, the frontend appears fine when using Google Chrome. I guess this is just dependant on different browsers but wanted to bring it up. This change will be in scope for another ticket and is out of scope for this task.

**Comment by sehersarin** [comment]
Hi Amy, 

Thank you for completing this PR - it looks great. I have completed the following steps to test this Story:

1. Ran the code locally and ensured no complications when starting up
a. Used student credentials and checked to make sure that there is a Cancel button next to each upcoming appointment for the student.
b. Used worker credentials and checked to make sure that there is a Cancel button next to each upcoming appointment for the student.
c. Clicked the button and ensured that the response is as expected.

I have also run the automated tests locally and ensured that all tests pass and have 100% file coverage for this story.

**Comment by sehersarin** [comment]
I am ready to approve this PR and will be merging it into the master branch!

**Review by sehersarin — APPROVED** [review]

**Pull request merged**